### PR TITLE
feat: add generics to more methods in Relation stub

### PR DESCRIPTION
**Pull request by rene-bos**
- [x] Added or updated tests
- [x] Documented user facing changes

**Changes**

Add generics to more public methods in the Relation stub file.

**Breaking changes**

None.

### CODE REVIEW

**Review by szepeviktor — APPROVED** [review]

File: tests/Type/data/relation.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (canvural) [inline tests/Type/data/relation.php:1]: We already have `model-relations.php` so we can just move these to that file.

**Review by canvural — CHANGES_REQUESTED** [review]
Overall looks ok. But I have a question about `getQuery` Would it return the custom query builder, if the model uses one? If so, we might need a new extension that handles that.

File: tests/Type/data/relation.php (hunk 1 above)
Comment (rene-bos) [reply tests/Type/data/relation.php]: Fixed.

**Comment by rene-bos** [comment]
> Overall looks ok. But I have a question about `getQuery` Would it return the custom query builder, if the model uses one? If so, we might need a new extension that handles that.

@canvural Yeah I think it will. The relation will create a new instance of the related class, and will call `newQuery()` on that. I think this change is already an nice improvement for most setups, are you fine with merging this first and extending it with an extension for custom query builders when somebody has time or the need for it?

**Review by canvural — APPROVED** [review]
Looks good 👍🏽 But we'd need to iterate on it to support custom builders.

**Comment by canvural** [comment]
Thank you!

**Pull request merged**